Add  Bitcoin Core v30 + standalone sv2-tp in integration tests

xyephy opened the pull request:
Use Bitcoin Core v30 + standalone sv2-tp in integration tests
Verified
[a218359](https://github.com/xyephy/sv2-apps/commit/a218359759832373fdb9ddd8ec4c66613c16cceb)
Implements the approach suggested  in https://github.com/stratum-mining/sv2-apps/issues/57: use bitcoin-node
with -ipcbind=unix and the standalone sv2-tp binary that connects via IPC.

Changes:
- Download official Bitcoin Core v30 binaries from bitcoincore.org
- Use bitcoin-node from libexec/ directory (not bitcoind)
- Launch bitcoin-node with -ipcbind=unix for IPC support
- Download standalone sv2-tp v1.0.3 that connects via IPC
- Support macOS (arm64/x86_64) and Linux (x86_64/aarch64)
- Add macOS binary signing to prevent security warnings

Test Results:
- template_provider_integration: All 3 tests pass (tp_low_diff, tp_mid_diff, tp_high_diff)
- jd_tproxy_integration: Both tests pass (aggregated and non-aggregated modes)

Fixes https://github.com/stratum-mining/sv2-apps/issues/57

plebhash commented on integration-tests/lib/template_provider.rs:
```diff
@@ -89,77 +106,136 @@ impl TemplateProvider {
+        let datadir = conf.staticdir.as_ref().expect("staticdir should be set");
+        let network = if conf.network == "signet" { "-signet" } else { "-regtest" };
+
+        let sv2_tp_process = Command::new(&sv2_tp_bin)
```
don't we need the `ipcconnect` parameter?

or is it being resolved to some default value that happens to match what `bitcoin-node` does?

plebhash replied:
oh I guess `datadir` takes care of that?

Sjors commented:
Might be better to leave the test for #58 to a followup.

plebhash commented:
@xyephy can we also update the `README.md`?

more specifically, this paragraph:

https://github.com/stratum-mining/sv2-apps/blob/dcfeea1d793167f78de517ae3fd02a6f87cfd70b/integration-tests/README.md?plain=1#L11-L14

not only the reference to Sjors fork will become outdated, I also notice that it's only mentioning `regtest`, while `signet` has also been supported since https://github.com/stratum-mining/stratum/pull/1760

xyephy commented:
> @xyephy can we also update the `README.md`?
> 
> more specifically, this paragraph:
> 
> https://github.com/stratum-mining/sv2-apps/blob/dcfeea1d793167f78de517ae3fd02a6f87cfd70b/integration-tests/README.md?plain=1#L11-L14
> 
> not only the reference to Sjors fork will become outdated, I also notice that it's only mentioning `regtest`, while `signet` has also been supported since [stratum-mining/stratum#1760](https://github.com/stratum-mining/stratum/pull/1760)

Yes, I will update it

xyephy commented:
@plebhash I have pushed updates to `README.md`. Let me know if it's better. Thanks.

GitGab19 reviewed (approved):
ACK.

Since you're touching ITF, could you please remove the `#![cfg(feature = "sv1")]` we have at line 1 of `integrations-tests/tests/sv1.rs`?

We don't have features in the `integration_tests_sv2` crate, and we are getting a warning for that.

xyephy commented:
> Thank you @xyephy.
> 
> The PR is not update with `main` btw, you have to rebase it.

let me do that, as I wait for CI tests and fix any  issues that might arise.

plebhash commented:
the removal of `sv1` flag is breaking unrelated stuff, we should probably address this on a separate issue/PR https://github.com/stratum-mining/sv2-apps/issues/69

for now, I'd leave it untouched

@xyephy sorry, can you revert those changes?

xyephy commented on integration-tests/lib/template_provider.rs:
```diff
@@ -1,32 +1,56 @@
 #[derive(Debug)]
 pub struct TemplateProvider {
     bitcoind: Node,
+    sv2_tp_process: Option<Child>,
```
Good point, the `.expect()` does panic on spawn failure right now, so we don't hit a graceful `None` case today. The `Option` is more for future extensibility (e.g., optional spawning via test config, or mocks in CI) and to keep the `Drop` impl non-panicky without extra checks.If it feels overkill for this PR, I'm happy to simplify to a plain `Child` (remove `Option` and adjust init/Drop accordingly).

Shourya742 replied:
The reason I’m saying this is that without sv2_tp, the TemplateProvider serves no purpose. To keep the structure conceptually clean, let’s remove the Option.

xyephy replied:
@Shourya742 I've updated, you can give it one last look.

xyephy commented:
> > Can you log an issue, to update the template provider in IT and remove manual download of bitcoin-core, once the corepc supports 30.0+

@Shourya742 had a look at  corepc logged issues  and turns out the work was already underway! 

- PR #387 (merged Oct 15): Adds v30 binary/RPC support. https://github.com/Rust-Bitcoin/corepc/pull/387
- Issue #384 (open): Tracks RPC changes for full v30 compatibility. https://github.com/Rust-Bitcoin/corepc/issues/384

This means we can drop the manual downloads here once corepc v0.11.0 releases.

xyephy commented:
> @xyephy can you address this point: [#61 (comment)](https://github.com/stratum-mining/sv2-apps/pull/61#discussion_r2494566784). After that you can rebase the PR on top of `main` and we can merge it.

@GitGab19 I've addressed it and rebased.
